test/contract/deployment/deployment_test.go

```diff
@@ -35,7 +35,7 @@ func TestListDeployments(t *testing.T) {
 				WithName(serverlessName).WithAtlasName(serverlessName)))
 		testProjectID, err := ch.ProjectID(ctx, projectName)
 		require.NoError(t, err)
-		ds := deployment.NewAtlasDeployments(ch.AtlasClient().ClustersApi, ch.AtlasClient().ServerlessInstancesApi, nil, false)
+		ds := deployment.NewAtlasDeployments(ch.AtlasClient().ClustersApi, ch.AtlasClient().ServerlessInstancesApi, nil, ch.AtlasClientSet().SdkClient20241113001.FlexClustersApi, false)
```

Comment:
q: should we replace serverless test already? I guess it is fine sine serverless is to be phased out soon and this is just the contract test which is mostly forward looking.

Reply:
We can probably leave the serverless tests in until the API is actually removed. Do you think there's a reason to get rid of them earlier?